include/nrf_inbuilt_key.h

```diff
@@ -0,0 +1,161 @@
+ *                            referred to in setting up a BSD Secure Socket or
+ *                            to manage the credential using nrf_key_mgmt module
+ *                            to read/delete/search the key.
+ * @param[in]  cred_type      Type of credential being created and stored for
+ *                            later use.
+ * @param[in]  p_buffer       Buffer containing the credential data.
+ * @param[in]  buffer_len     Length of the buffer holding the credential data.
+ *
+ * @retval 0        If create operation was successful.
+ * @retval EIO      If operation was not successful due to internal error.
+ * @retval ENOBUFS  If the operation could not be performed because it could
+ *                  not allocate enough intermediate buffers to perform the
+ *                  operation.
+ * @retval ENOENT   If the sec_tag indicated cannot be written.
+ * @retval ENOMEM   If there was not memory enough to store the credential data.
+ * @retval EPERM    If the application did not have permission to do the
+ *                  operation.
+ * @retval EACCES   If the operation could not be performed while modem is
+ *                  in active state.
+ * @retval EINVAL   If one or more of the provided parameters are not valid.
+ */
+int nrf_inbuilt_key_write(nrf_sec_tag_t sec_tag,
+			  nrf_key_mgnt_cred_type_t cred_type, uint8_t *p_buffer,
+			  u16_t buffer_len);
+
+
+/**@brief Read back a credential from persistent storage.
+ *
+ * This function will read the credential from persistent memory referenced by a
+ * sec_tag.
+ *
+ * @param[in]     sec_tag       Application defined tag for this credential to
+ *                              read.
+ * @param[in]     cred_type     Type of credential being read.
+ * @param[in]     p_buffer      Output buffer containing where to write the read
+ *                              credential data.
+ * @param[inout]  p_buffer_len  Length of the output buffer as input, and length
+ *                              used as output parameter.
+ *
+ * @retval 0        If read operation was successful.
+ * @retval EIO      If operation was not successful due to internal error.
+ * @retval ENOBUFS  If the operation could not be performed because it could
+ *                  not allocate enough intermediate buffers to perform the
+ *                  operation.
+ * @retval ENOENT   If there was no credential associated with the sec_tag
+ *                  and cred_type.
+ * @retval EPERM    If the application did not have permission to do the
+ *                  operation.
+ * @retval EINVAL   If provided buffer is to small for result data. If
+ *                  failing with this error, the size needed is provided as
+ *                  output parameter by reference in the output p_buffer_len
+ *                  output parameter.
+ */
+int nrf_inbuilt_key_read(nrf_sec_tag_t sec_tag,
+			 nrf_key_mgnt_cred_type_t cred_type, uint8_t *p_buffer,
```

Comment:
My intention is not to change any API or add any new API. I aim to make a replacement of what is already used. When removing the the functions in an upcomming bsdlib release, it will be easier for users to migrate if we keep the API close to what it was in the beginning. The compare function is a great suggestion, but i'm not intending to add this in this PR.